sophie_pub.snakefile

```diff
@@ -0,0 +1,233 @@
+# import all of the sophie libraries.
+# I chose to run all of the python sophie code via the "run" directive, which doesn't allow for the use of an environment.
+# Therefore, all of the dependencies for sophie need to be installed in the environment in which this workflow will be executed.
+
+import os
+import sys
+import glob
+import pickle
+# for mac, import matplotlib and change backend
+import matplotlib
+matplotlib.use('TkAgg')
+import pandas as pd
+import numpy as np
+import scipy.stats as ss
+from sklearn import preprocessing
+from keras.models import load_model
+from ponyo import utils, train_vae_modules, simulate_expression_data
+from generic_expression_patterns_modules import process, stats, ranking, new_experiment_process
+
+# I have removed the input and output files from the sophie config file.
+# Instead, these files are solved through wildcards, and their paths are specified as inputs and outputs for each of the snakemake rules.
+STRAIN = ["pa14", "pao1"]
+PUB_COMPARISONS = ['pub-vs-lbpao1', 'pub-vs-lbpa14', 'pub-vs-m63']
+
+# constrain these wildcards so they solve properly
+wildcard_constraints:
+    run_id = "\d+" # constrain to digits only
+
+# A few of the config-specified variables also need to be accessible as global variables in the snakefile.
+# They're used to name input/output files correctly.
+# I read them in below using sophie's config file reader.
+# NUM_SIMULATED_RUNS also needed to be specified as a range from 0:num_simulated_runs to properly solve files names, so that is specified below.
```

Comment:
Where? I don't see it

Reply:
```suggestion
# NUM_SIMULATED_RUNS (e.g. RUN_IDS below) also needed to be specified as a range from 0:num_simulated_runs to properly solve files names, so that is specified below.
```